### PR TITLE
Enable upload package cache to Azure Storage

### PR DESCRIPTION
**Pull request by troydai**
@pranavkm @javiercn

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @javiercn

File: src/DependenciesPackager/Program.cs [hunk 1]
```diff
@@ -174,6 +178,11 @@ static int Main(string[] args)
                 "The prefix to use for the zip file name",
                 CommandOptionType.SingleValue);
 
+            var uploadToAzure = app.Option(
+                "--upload-to-azure",
```
Comment (pranavkm) [inline src/DependenciesPackager/Program.cs]: Seems kinda out of place with the rest of the stuff this thing does. Would it make sense to have a different task that does the upload?

File: src/DependenciesPackager/Program.cs [hunk 2]
```diff
@@ -702,6 +719,31 @@ private int RunDotnetRestore(string projectJson, string packagesFolderName)
...
         }
 
+        private async Task UploadPackagesCache(string zipFilePath)
+        {
+            if (string.IsNullOrEmpty(zipFilePath) || !File.Exists(zipFilePath))
+            {
+                throw new ArgumentException(nameof(zipFilePath));
+            }
+
+            var connectionString = Environment.GetEnvironmentVariable("AZURE_STORAGE_ACCOUNT");
```
Comment (pranavkm) [inline src/DependenciesPackager/Program.cs]: Have these be passed in (instead of env variables).

**Comment by pranavkm** [comment]
:watch: for the question about splitting the tool up.

File: src/DependenciesPackager/Program.cs (hunk 1 above)
Comment (troydai) [inline src/DependenciesPackager/Program.cs]: Yes we can do that. The con is 
1) Bigger makefile.proj  to accommodate more tasks and conditions;
2) Need to ensure zip file name is in sync between tasks.

**Comment by troydai** [comment]
@pranavkm move everything to its own project.

**Comment by pranavkm** [comment]
:shipit:

**Pull request merged**